R/QueueDataWarehouse.R

```diff
@@ -1,164 +1,206 @@
+#' 
+#' All data warehouse results are paged in chunks of 20 MB. Add "page": to \code{Report.Get} 
+#' to determine the page returned. If no page is specified then the first page is returned.
+#'
+#' @description A QueueDataWarehouse report is a report where metrics are
+#' retrieved, broken down by an unlimited number of elements such as page, eVar, prop, etc, and
+#' with or without temporal aggregation.
+#'
+#' @title Queue a DataWarehouse Report
+#'
+#' @param reportsuite.id Report suite id
+#' @param date.from Start date for the report (YYYY-MM-DD)
+#' @param date.to End date for the report (YYYY-MM-DD)
+#' @param metrics List of metrics to include in the report
+#' @param elements List of elements to include in the report
+#' @param classification List of SAINT classifications for each element. If supplied, must be a character
+#' vector of length equal to elements.
+#' @param date.granularity Time granularity of the report (year/month/week/day/hour), default to 'day'. Pass
+#' a blank string or NA if you do not want any date granularity.
+#' @param segment.id Id of Adobe Analytics segment to retrieve the report for
+#' @param data.current TRUE or FALSE - whether to include current data for reports that include today's date
+#' @param expedite Set to TRUE to expedite the processing of this report
+#' @param interval.seconds How long to wait between attempts
+#' @param max.attempts Number of API attempts before stopping
+#' @param validate whether to submit report definition for validation before requesting the data.
+#' @param enqueueOnly only enqueue the report, don't get the data. returns report id, which you can later use to get the data
+#' @param ftp FTP client parameters, only used if enqueueOnly=TRUE. Double check ftp parameters before requesting
+#' a long report.
+#'
+#' @importFrom jsonlite toJSON unbox
+#' @importFrom utils read.csv
+#'
+#' @return Data frame or report id, if enqueueOnly is TRUE
+#'
+#' @examples
+#' \dontrun{
+#' report.data <- QueueDataWarehouse("your_report_suite",
+#'                             "2014-01-01",
+#'                             "2014-01-07",
+#'                             c("visits", "pageviews","event10"),
+#'                             c("page","geoCountry","geoCity"),
+#'                             enqueueOnly=TRUE,
+#'                             ftp = list(host = "myftpserver.com",
+#'                                        port = "21",
+#'                                        directory = "/fromDW/",
+#'                                        username = "memyselfandirene",
+#'                                        password = "valkilmer",
+#'                                        filename = "myreport.csv")
+#'                             )
+#'}
+#' @export
+#' 
+QueueDataWarehouse <- function(reportsuite.id, date.from, date.to, metrics, elements,
+                                   classification = c(), date.granularity='day', segment.id='',

```

Comment:
Even though these arguments are optional, I think you need to put `classification` at the end of the parameter list, as we can't guarantee people aren't using the list of parameters as it exists positionally

Reply:
Good point. Any thoughts on updating doc for interval.seconds to reflect the default of 5?

Reply:
Sure, updating the docs are always fine. Is it in the docstring that needs to be changed or the RSiteCatalyst HTML docs?

Reply:
Only looked in the Rd.